universite-du-quebec-a-montreal.csl

```diff
@@ -276,218 +337,342 @@
+        </group>
+      </else-if>
+      <else-if type="map" match="any">
+        <choose>
+          <if match="any" variable="container-title">
+            <group delimiter=" ">
+              <text variable="title"/>
+              <text variable="genre" prefix="[" suffix="]"/>
+            </group>
+            <group delimiter=" : " prefix=". ">
+              <text term="scale"/>
+              <text variable="scale"/>
+            </group>
+          </if>
+          <else>
+            <text variable="title" font-style="italic"/>
+            <text variable="archive" prefix=". " suffix="."/>
+            <text variable="genre" prefix=" [" suffix="]"/>
+            <group delimiter=" : " prefix=". ">
+              <text term="scale"/>
+              <text variable="scale"/>
+            </group>
+            <text variable="number-of-pages" prefix=". "/>
+          </else>
+        </choose>
+      </else-if>
+      <else-if type="figure" match="any">
+        <choose>
+          <if match="any" variable="container-title">
+            <group delimiter=" ">
+              <text variable="title"/>
+              <text variable="genre" prefix="[" suffix="]"/>
+            </group>
+          </if>
+          <else>
+            <group delimiter=" ">
+              <text variable="title" font-style="italic"/>
+              <text variable="genre" prefix="[" suffix="]"/>
+            </group>
+          </else>
+        </choose>
+      </else-if>
+      <else-if type="interview" match="all" variable="container-title">
+        <group delimiter=". ">
+          <names variable="interviewer">
+            <label form="verb" text-case="capitalize-first" suffix=" "/>
+            <name and="text" delimiter-precedes-last="never" initialize-with="."/>
+          </names>
+          <text variable="title"/>
+        </group>
+      </else-if>
+      <else-if type="post post-weblog" match="any" variable="container-title">
         <group delimiter=". ">
           <text variable="number" font-style="italic" prefix="Bill " suffix=": "/>
-          <text variable="title" font-style="normal"/>
+          <text variable="title"/>
+        </group>
+      </else-if>
+      <else>
+        <group delimiter=". ">
+          <text variable="title" font-style="italic"/>
+          <text variable="archive"/>
+          <text variable="genre" prefix="[" suffix="]"/>
         </group>
-        <text variable="genre" prefix=". [" suffix="]"/>
       </else>
     </choose>
   </macro>
   <macro name="publisher">
     <choose>
-      <if type="report" match="any">
+      <if type="report thesis" match="any">
         <group delimiter=" : ">
           <text variable="publisher-place"/>
           <text variable="publisher"/>
         </group>
       </if>
-      <else-if type="thesis" match="any">
-        <group delimiter=": ">
+      <else-if type="speech" match="any">
+        <group delimiter=", ">
+          <text variable="abstract"/>
```

Comment:
This is odd -- why would you add the whole abstract for a speech/presentation. I'd reconsider that.

Reply:
I agree. This was for Mendeley as there is no field mapped to event variable. I will change this for publisher variable.